pkg/translator/azurelogs/resourcelogs_to_logs.go

```diff
@@ -91,6 +95,18 @@ func (r ResourceLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
 		return plog.Logs{}, fmt.Errorf("JSON parse failed: %w", iter.Error)
 	}
 
+	// Re-parse to capture raw record JSON for fields not in the struct
+	var rawRecords struct {
+		Records []json.RawMessage `json:"records"`
+	}
+	if err := json.Unmarshal(buf, &rawRecords); err == nil {
+		for i := range azureLogs.Records {
```

Comment:
Could we use [maps.Copy](https://pkg.go.dev/maps#Copy) here?

Reply:
Hmm I dont think so. `maps.Copy` is for maps but here we are copying slices.